Assets/Scripts/Entity/Unit/Unit.cs

```diff
@@ -2,28 +2,54 @@
 using System.Collections.Generic;
 using UnityEngine;
 using UnityEngine.UI;
+using UnityEngine.AI;
 
 /*
     Class that defines a GameObject attached to a unit structure.
  */
 public class Unit : MonoBehaviour
 {
     public UnitEntity unitEntity;
+    public GameObject targettedEntity;
+    private float attackCooldown = 0;
+    public bool isAttacking = false;
+    
 
     // Start is called before the first frame update
     void Start()
     {
+        attackCooldown = unitEntity.attackTime;
     }
 
     // Update is called once per frame
-    void Update()
+    void FixedUpdate()
     {
+        reduceAttackTimeTick();
```

Comment:
- The attack behavior could be put in its own function.
- Also, other types of entities have the attack behavior. Delegation to this behavior or inheritance from another class could be beneficial